### PR TITLE
Proposal size check

### PR DESCRIPTION
**Pull request by eerkaijun**
https://github.com/category-labs/category-internal/issues/796
https://github.com/category-labs/category-internal/issues/1016

i feel like we only need to enforce proposal size check when creating proposal as it's for making sure proposal can be broadcasted in time, and no need to validate that in eth-block-validator, but let me know if people think otherwise

### CODE REVIEW

File: monad-consensus-types/src/payload.rs [hunk 1]
```diff
@@ -48,6 +48,8 @@ impl AsRef<[u8]> for Gas {
 
 pub const BASE_FEE_PER_GAS: u64 = 1_000;
 pub const PROPOSAL_GAS_LIMIT: u64 = 150_000_000;
+/// Max proposal size in bytes (average transactions ~400 bytes)
+pub const PROPOSAL_SIZE_LIMIT: u64 = 5_000_000;
```
Comment (eerkaijun) [inline monad-consensus-types/src/payload.rs]: we can switch this value

File: monad-eth-txpool/src/tracked/mod.rs [hunk 2]
```diff
@@ -232,6 +233,14 @@ where
                 return TrackedTxHeapDrainAction::Skip;
             }
 
+            if total_size
+                .checked_add(tx.size().into())
+                .expect("total size does not overflow")
```
Comment (eerkaijun) [inline monad-eth-txpool/src/tracked/mod.rs]: there's a bunch of expect in this file, we should remove them

Comment (eerkaijun) [reply monad-eth-txpool/src/tracked/mod.rs]: removed some, we should review the remaining ones

File: monad-consensus-types/src/payload.rs (hunk 1 above)
Comment (omegablitz) [reply monad-consensus-types/src/payload.rs]: let's set to 4MB, so 4_000_000

**Comment by eerkaijun** [comment]
ok sounds good, added

**Pull request merged**